stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2473,6 +2473,11 @@ collections = [
 /// keys, followed by the values.
 //
 // See the docs at the top of the file for more details on this type
+//
+// NOTE: The precise layout of this type is relied on in the runtime
+// to provide a statically allocated empty singleton. 
```

Comment:
Fixed-layout or not won't make a difference here. I think if the layout algorithm changes and doesn't match what the compiler expects, everything will be broken, and we'll know right away. So this PR looks OK to me.